feat: support huaweicloud OBS for Oss

showlovetommy opened the pull request:
[feat: support huaweicloud OBS for Oss ](https://github.com/apitable/apitable/issues/499)

showlovetommy committed:
extend oss for huaweicloud obs

showlovetommy committed:
extend oss for huaweicloud obs

showlovetommy committed:
extend oss for huaweicloud obs

showlovetommy committed:
extend oss for huaweicloud obs

Commit:
create testclass for HuaweiCloudOBSClient Test

Commit:
create testclass for HuaweiCloudOBSClient Test

Commit:
create testclass for HuaweiCloudOBSClient Test

Commit:
create testclass for HuaweiCloudOBSClient Test

Commit:
Merge remote-tracking branch 'origin/feature-obj-storage-huaweicloudobs' into feature-obj-storage-huaweicloudobs

# Conflicts:
#	backend-server/lib.gradle

Commit:
create testclass for HuaweiCloudOBSClient Test

Commit:
create testclass for HuaweiCloudOBSClient Test

Commit:
create testclass for HuaweiCloudOBSClient Test

showlovetommy committed:
Merge branch 'apitable:develop' into feature-obj-storage-huaweicloudobs

Commit:
feat: add huaweicloud obs configuration to application.yml

Commit:
feat: add huaweicloud obs configuration to application.yml

Commit:
feat: add huaweicloud obs configuration to application.yml

Commit:
feat: override HuaweiCloudOssClientRequest methods <getStatObject>, <executeStreamFunction>

Commit:
feat: override HuaweiCloudOssClientRequest methods <getStatObject>, <executeStreamFunction>

Commit:
feat: override HuaweiCloudOssClientRequest methods <getStatObject>, <executeStreamFunction>

Commit:
feat: override HuaweiCloudOssClientRequest methods <getStatObject>, <executeStreamFunction>

Commit:
feat: override HuaweiCloudOssClientRequest methods <getStatObject>, <executeStreamFunction>

Commit:
feat: override HuaweiCloudOssClientRequest methods <getStatObject>, <executeStreamFunction>

Commit:
feat: add huaweicloudobs env_var on .env

Commit:
feat: add huaweicloudobs env_var on .env

Commit:
feat: add huaweicloudobs env_var on .env

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/apitable/apitable?pullRequest=500) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/apitable/apitable?pullRequest=500) before we can accept your contribution.<br/>**5** out of **6** committers have signed the CLA.<br/><br/>:white_check_mark: showlovetommy<br/>:white_check_mark: buzuosheng<br/>:white_check_mark: laboonly<br/>:white_check_mark: yort-feng<br/>:white_check_mark: boris-w<br/>:x: mmy<br/><hr/>**mmy** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/apitable/apitable?pullRequest=500) it.</sub>

networkhermit commented on .env:
```diff
@@ -99,6 +99,10 @@ AWS_ACCESS_SECRET=apitable@com
 
+HUAWEICLOUD_ACCESS_KEY=YOUR AK
+HUAWEICLOUD_SECRET_KEY=YOUR SK
+HUAWEICLOUD_ENDPOINT=YOUR ENDPOINT
```
I think HUAWEICLOUD_OBS_* is better than HUAWEICLOUD_*, to avoid confusion about the scope of the api key.

showlovetommy replied:
thanks for your suggestion, i will fix this

showlovetommy replied:
> I think HUAWEICLOUD_OBS_* is better than HUAWEICLOUD_*, to avoid confusion about the scope of the api key.
 I have rename these env_vars, below:
[fix: rename the huaweicloud obs env_var on .env & application.yml](https://github.com/apitable/apitable/pull/500/commits/c40206256b69f0151ffe7f9f605e2d50b4aacacf)

buzuosheng committed:
hpm log with api prefix  (#476)

When web-server runs, the previous log looked like this:
```
Type-checking in progress...
[HPM] Proxy created: /  -> http://127.0.0.1:3333/
[HPM] Proxy created: /  -> http://127.0.0.1:8081/
[HPM] Proxy created: /  -> http://127.0.0.1/
[HPM] Proxy created: /room  -> http://127.0.0.1:3005/
[HPM] Proxy created: /notification  -> http://127.0.0.1:3002/
> Ready on http://localhost:3000/
```
Now it is: 
``` 
Type-checking in progress...
[HPM] Proxy created: /nest  -> http://127.0.0.1:3333/
[HPM] Proxy created: /api  -> http://127.0.0.1:8081/
[HPM] Proxy created: /fusion  -> http://127.0.0.1/
[HPM] Proxy created: /room  -> http://127.0.0.1:3005/
[HPM] Proxy created: /notification  -> http://127.0.0.1:3002/
> Ready on http://localhost:3000/
```

Co-authored-by: Kilian <wangkailang@users.noreply.github.com>

showlovetommy committed:
Merge branch 'apitable:develop' into feature-obj-storage-huaweicloudobs

showlovetommy committed:
fix: rename the huaweicloud obs env_var on .env & application.yml

showlovetommy committed:
fix: rename the huaweicloud obs env_var on .env & application.yml

showlovetommy committed:
Merge remote-tracking branch 'origin/feature-obj-storage-huaweicloudobs' into feature-obj-storage-huaweicloudobs

# Conflicts:
#	.env
#	backend-server/application/src/main/resources/application.yml
#	backend-server/lib.gradle

shawndenggh commented:
Thanks for your contribution, I will review the code now @showlovetommy

shawndenggh commented on backend-server/lib.gradle:
```diff
@@ -1,151 +1,152 @@
-            'shedlock'                     : "net.javacrumbs.shedlock:shedlock-spring:${shedlock_version}",
-            'shedlock-redis'               : "net.javacrumbs.shedlock:shedlock-provider-redis-spring:${shedlock_version}",
-    ]
+  dependencies = [
```
It seems that the code style you use is different from us. your IDE tool has changed the indentation of this file, please use our code style file to format it? this configuration file is placed in the directory named: `apitable/backend-server/config/checkstyle/checkstyle.xml`. Thanks

showlovetommy replied:
> It seems that the code style you use is different from us. your IDE tool has changed the indentation of this file, please use our code style file to format it? this configuration file is placed in the directory named: `apitable/backend-server/config/checkstyle/checkstyle.xml`. Thanks

Sorry,but this might not be my wrong.The others *.gradle files have the same Tab Size like my local.
And I have checked this file's history. It seems like the TAB Size or Indent change in this version below.

https://github.com/apitable/apitable/blob/256b416b2c243f8e18d0b7dc1680879e29e89a8c/backend-server/lib.gradle

Only this [backend-server/lib.gradle](https://github.com/apitable/apitable/pull/500/files/94aa2a710dcdfa7699aea4c5d9e4a8f3a6e648de#diff-c9a3be93a00d562bd48e0bab34a829f11b5d7929a2bebbee4abc9a0123cb99a8)
format is different from the others *.gradle.  
Shall we take a check?

shawndenggh replied:
Are you using Intellij IDEA for coding？show me your groovy code style config under `Settings/Editor/Code Style/Groovy`, like below
![image](https://user-images.githubusercontent.com/39178385/223308260-d7d9ed6a-a6aa-48d8-a86b-35540f01ae7c.png)

showlovetommy replied:
Yes I am using Intellij IDEA. This is my config below.
![20230307105837](https://user-images.githubusercontent.com/21049427/223309154-9bff5859-91ac-4c68-86f6-c32525fb102d.png)

showlovetommy replied:
thanks for your help, I have fixed this IDE Setting!

shawndenggh commented on backend-server/shared/starters/oss/src/main/java/com/apitable/starter/oss/autoconfigure/OssProperties.java:
```diff
@@ -38,6 +38,8 @@ public class OssProperties {
 
     private Aliyun aliyun;
 
+    private HuaweiCloud huaweicloud;
```
the properties key `huaweicloud` can not match `huawei-cloud`, the  key name rules must be `lowerCamelCase`

showlovetommy replied:
the key name has been `lowerCamelCase`

shawndenggh commented on backend-server/shared/starters/oss/src/main/java/com/apitable/starter/oss/core/obs/HuaweiCloudOssClientRequest.java:
```diff
@@ -0,0 +1,309 @@
+  
+    private final ObsClient obsClient;
+    private boolean autoCreateBucket = false;
+    private MultipartState multipartState;
```
convert local variable since this variable used in `uploadStreamForObject` method only

showlovetommy replied:
variable `multipartState ` has convert to local and I have add comment to methods on `HuaweiCloudOssClientRequest.java`.

shawndenggh reviewed (commented):
it is very helpful for reading code If there is comment in detail. please help me understand your logic code.

Commit:
fix: apitable register bugfix

Commit:
fix: clear console.log

boris-w committed:
fix: the logo on the create space page supports the dark theme (#508)

showlovetommy committed:
Merge branch 'develop' into feature-obj-storage-huaweicloudobs

Commit:
fix: missing import

Commit:
sync: hosted cloud (#511)

Signed-off-by: wangkailang <wangkailang@vikadata.com>
Co-authored-by: Boris Bo Wang <wangbo@vikadata.com>
Co-authored-by: Bron ZiJing Liu <liuzijing@vikadata.com>
Co-authored-by: Aria <jianweisummer@gmail.com>
Co-authored-by: liuyi <liuyi@vikadata.com>
Co-authored-by: Kilian <wangkailang@users.noreply.github.com>
Co-authored-by: BoomBoommm <wuyitao@vikadata.com>
Co-authored-by: Chambers <chenbochao@vikadata.com>
Co-authored-by: plodsoft <xplzjwz@gmail.com>
Co-authored-by: 苏简 <JaneSu@AriadeMac-mini.local>
Co-authored-by: SkyHuang <huangchangpeng@vikadata.com>
Co-authored-by: Kelly Chan <kelly@apitable.com>

showlovetommy committed:
Merge branch 'apitable:develop' into feature-obj-storage-huaweicloudobs

showlovetommy committed:
fix: fix the lib.gradle format ,convert local variable and add some methods comment

showlovetommy committed:
fix: fix the lib.gradle format ,convert local variable and add some methods comment

showlovetommy committed:
Merge remote-tracking branch 'origin/feature-obj-storage-huaweicloudobs' into feature-obj-storage-huaweicloudobs

# Conflicts:
#	backend-server/lib.gradle

shawndenggh commented on .gitignore:
```diff
@@ -77,3 +77,5 @@ scripts/design_token/__pycache__
 
 # eslint results
 eslint-results.sarif
+.vscode/settings.json
```
please do not modify project gitignore file content.
